fc/parsing/actions.py

```diff
@@ -521,47 +523,101 @@ def _expr(self):
         return self
 
 
-class InputVariable(BaseGroupAction):
-    """
-    Parse action for input variables defined in the model interface.
+class VariableReference:
+    """Mixin providing properties for resolving variable references.
+
+    Used by input & output variable specifications, inter alia.
+
+    Properties:
+
+    ``prefixed_name``
+        A 'prefix:local_name' string.
+
+    ``ns_prefix``
+        The namespace prefix part of ``prefixed_name``.
+
+    ``local_name``
+        The local name part of ``prefixed_name``.
+
+    ``ns_uri``
+        Once namespace prefixes have been resolved, the namespace URI corresponding to ``ns_prefix``.
+
+    ``rdf_term``
+        The RDF term that annotates variable(s) we reference.
 
-    ``input <prefix:term> [units <uname>] [= <initial_value>]``
     """
     def _expr(self):
         name = self.prefixed_name = self.get_named_token_as_string('name')
         self.ns_prefix, self.local_name = name.split(':', 1)
         self.ns_uri = None  # Will be set later using protocol's namespace mapping
+        return self
+
+    @property
```

Comment:
This seems really hacky to me!
It's a property that will give some kind of error if I use it before the parent object is in a certain state?
Let's just make it a property that's either set or not, or a method that can raise an exception

Reply:
Well, all the users are in this file and it only gets accessed in the context of manipulating a model, by which point it *is* set. The docstring was just a warning if in the future it starts being used elsewhere.

Reply:
:D That doesn't make it better!